sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/exemplar/FixedSizeExemplarReservoir.java

```diff
@@ -47,18 +52,32 @@ public void offerLongMeasurement(long value, Attributes attributes, Context cont
 
   @Override
   public void offerDoubleMeasurement(double value, Attributes attributes, Context context) {
+    if (storage == null) {
+      storage = initStorage();
+    }
     int bucket = reservoirCellSelector.reservoirCellIndexFor(storage, value, attributes, context);
     if (bucket != -1) {
       this.storage[bucket].recordDoubleMeasurement(value, attributes, context);
       this.hasMeasurements = true;
     }
   }
 
+  private ReservoirCell[] initStorage() {
+    ReservoirCell[] storage = new ReservoirCell[this.size];
+    for (int i = 0; i < size; ++i) {
+      storage[i] = new ReservoirCell(this.clock);
+    }
+    return storage;
+  }
+
   @Override
   public List<T> collectAndReset(Attributes pointAttributes) {
     if (!hasMeasurements) {
       return Collections.emptyList();
     }
+    if (storage == null) {
+      storage = initStorage();
+    }
```

Comment:
Storage will have been initialized if the reservoir received measurements. No need to initialize it here if it hasn't received any measurements.
```suggestion
    if (!hasMeasurements || storage == null) {
      return Collections.emptyList();
    }
```